code/smoothie.js

```diff
@@ -0,0 +1,314 @@
+const randomSelector = array => {
+  return array[Math.floor(Math.random() * array.length)]
+}
+
+// Function that toggles loading-gif
+const toggleLoader = (whatBox) => {
+  document.getElementById(whatBox).classList.add('loading')
+  // remove the loading-gif...
+  const removeLoader = () => {
+    document.getElementById(whatBox).classList.remove('loading')
+  }
+  //... after 1.5 second
+  setTimeout(removeLoader, 1500)
+}
+
+// 3 x Display random image and title from each category 
+// + add value to smoothie-object
+// + check if smoothie-button color should be changed
+const pickMainFruit = () => {
+  mainFruit = randomSelector(mainFruits)
+
+  // Pick a random fruit, unless favorite is selected 
+  // (then pick random from mainFruitsPlusFavorite-array)
+  if (mainFruitsPlusFavorite[6] == undefined) {
+    mainFruit = randomSelector(mainFruits)
+  } else {
+    mainFruit = randomSelector(mainFruitsPlusFavorite)
+  }
+
+  // Change titile of Main Fruit to reveal what the fruit is
+  const toggleMainFruitTitle = () => {
+    document.getElementById('fruit-header').innerHTML = `Main Fruit/berry: ${mainFruit.name}`
+  }
+
+  toggleLoader('main-fruit-box')
+  // Delay the reveal of name with same as loader
+  setTimeout(function () { toggleMainFruitTitle(); }, 1500)
+
+  document.getElementById('main-fruit-box').style.background = `url('${mainFruit.image}')`
+
+  smoothie.mainFruit = mainFruit.name
+  colorCheck()
+}
+
+const pickMixer = () => {
+  mixer = randomSelector(mixers)
+
+  // Change titile of Main Fruit to reveal what the fruit is
+  const toggleMixerTitle = () => {
+    document.getElementById('mixer-header').innerHTML = `Mixer: ${mixer.name}`
+  }
+
+  toggleLoader('mixer-box')
+  setTimeout(function () { toggleMixerTitle(); }, 1500)
+
+  document.getElementById('mixer-box').style.background = `url('${mixer.image}')`
+
+  smoothie.mixer = mixer.name
+  colorCheck()
+}
+
+const pickSuperBooster = () => {
+  superBooster = randomSelector(superBoosters)
+
+  const toggleSuperBoosterTitle = () => {
+    document.getElementById('superbooster-header').innerHTML = `Super Booster: ${superBooster.name}`
+  }
+
+  toggleLoader('superbooster-box')
+  setTimeout(function () { toggleSuperBoosterTitle(); }, 1500)
+
+  document.getElementById('superbooster-box').style.background = `url('${superBooster.image}')`
+
+  smoothie.superBooster = superBooster.name
+  colorCheck()
+}
+
+// Randomize-button in the header - to quickly randomize a smoothie
+const randomizeSmoothie = () => {
+  pickMainFruit()
+  pickMixer()
+  pickSuperBooster()
+}
+
+// Check if the show smoothie-button should change into 'active' color
+// (based on if all object properties of smoothie-object is set)
+const colorCheck = () => {
+  if (smoothie.mainFruit !== undefined && smoothie.mixer !== undefined && smoothie.superBooster !== undefined) {
+    document.getElementById('mixer-button').classList.add('active')
+  } else {
+  }
```

Comment:
Empty else?

Reply:
I could suggestion the function to be renamed `checkColor`
Conforming to the convention of naming functions as `verbThing` e.g. `isValid`, `showModal`, `hideButton`